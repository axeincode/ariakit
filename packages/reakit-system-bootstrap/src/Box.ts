import { css, cx } from "emotion";
import { BoxProps } from "reakit/Box/Box";
import {
  PaletteBoxOptions,
  useBoxProps as usePaletteBoxProps
} from "reakit-system-palette/Box";

export type BootstrapBoxOptions = PaletteBoxOptions;

export function useBoxProps(
  { unstable_system }: BootstrapBoxOptions,
  htmlProps: BoxProps = {}
): BoxProps {
  const { style } = usePaletteBoxProps({ unstable_system });

  const box = css`
    box-sizing: border-box;
    font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI",
      "Helvetica Neue", Helvetica, Arial, sans-serif, "Apple Color Emoji",
      "Segoe UI Emoji", "Segoe UI Symbol";

    ${style as any}
  `;

  return { ...htmlProps, className: cx(box, htmlProps.className) };
}
