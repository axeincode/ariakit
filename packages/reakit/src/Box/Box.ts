import * as React from "react";
import { unstable_createComponent } from "../utils/createComponent";
import { useHook } from "../system/useHook";
import { Keys } from "../__utils/types";

export type unstable_BoxOptions = {
  /** Options passed to `reakit-system-*` */
  system?: any;
};

export type unstable_BoxProps = React.HTMLAttributes<any> &
  React.RefAttributes<any>;

export function useBox(
  options: unstable_BoxOptions = {},
  htmlProps: unstable_BoxProps = {}
) {
  htmlProps = useHook("useBox", options, htmlProps);
  return htmlProps;
}

const keys: Keys<unstable_BoxOptions> = ["system"];

useBox.__keys = keys;

export const Box = unstable_createComponent({
  as: "div",
  useHook: useBox
});
