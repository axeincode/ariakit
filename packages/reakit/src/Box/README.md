---
path: /docs/box
redirect_from:
  - /components/box
  - /components/block
  - /components/flex
  - /components/grid
  - /components/inline
  - /components/inlineblock
  - /components/inlineflex
  - /components/avatar
  - /components/blockquote
  - /components/card
  - /components/card/cardfit
  - /components/code
  - /components/heading
  - /components/image
  - /components/link
  - /components/list
  - /components/navigation
  - /components/paragraph
  - /components/table
  - /components/table/tablewrapper
---

# Box

Hello, "World"!

## Usage

```jsx
import { Box } from "reakit";

function Example() {
  return <Box>Box</Box>;
}
```

## Props

<!-- Automatically generated -->

### `Box`

No props to show
