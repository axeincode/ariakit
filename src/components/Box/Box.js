import styled from 'styled-components'
import { compose, setDisplayName, setPropTypes } from 'recompose'
import as from '../../enhancers/as'
import Base from '../Base'

const Box = styled(Base)`
  border: 1px solid rgba(0, 0, 0, 0.3);
  border-radius: 0.25em;
`

const enhance = compose(
  as('div'),
  setDisplayName('Box'),
  setPropTypes(Base.propTypes),
)

export default enhance(Box)
