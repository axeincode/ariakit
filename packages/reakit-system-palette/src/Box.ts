import { unstable_BoxOptions, unstable_BoxProps } from "reakit/Box/Box";
import { mergeProps } from "reakit/utils/mergeProps";
import { usePalette } from "./utils/palette";
import { useContrast } from "./utils/contrast";

export type PaletteBoxOptions = unstable_BoxOptions & {
  unstable_system: {
    palette?: string;
    fill?: "opaque" | "outline";
  };
};

export function useBoxProps(
  { unstable_system: system = {} }: PaletteBoxOptions,
  htmlProps: unstable_BoxProps = {}
) {
  const color = usePalette(system.palette);
  const contrast = useContrast(color);
  const textColor = system.fill === "opaque" ? contrast : color;
  const backgroundColor = system.fill === "opaque" ? color : undefined;
  const borderColor = system.fill === "outline" ? color : undefined;
  const style = {
    ...(textColor ? { color: textColor } : {}),
    ...(backgroundColor ? { backgroundColor } : {}),
    ...(borderColor ? { border: `1px solid ${borderColor}`, borderColor } : {})
  };

  return mergeProps({ style }, htmlProps);
}
