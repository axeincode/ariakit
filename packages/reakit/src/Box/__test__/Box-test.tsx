import * as React from "react";
import { render } from "react-testing-library";
import Box from "../Box";

test("html attrs", () => {
  const { getByTestId } = render(
    <Box id="test" aria-label="test" data-testid="test" />
  );

  expect(getByTestId("test")).toHaveAttribute("id", "test");
  expect(getByTestId("test")).toHaveAttribute("aria-label", "test");
});

test("styled", () => {
  const { container } = render(<Box />);
  expect(container.firstChild).toMatchInlineSnapshot(`
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`);
});

test("styled palette", () => {
  const { container } = render(<Box palette="primary" />);
  expect(container.firstChild).toMatchSnapshot(`
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: rgb(33, 150, 243);
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`);
});

test("styled opaque", () => {
  const { container } = render(<Box palette="primary" opaque />);
  expect(container.firstChild).toMatchSnapshot(`
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: rgb(33, 150, 243);;
  color: white;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`);
});

test("styled tone", () => {
  const { container } = render(<Box palette="primary" tone={1} opaque />);
  expect(container.firstChild).toMatchSnapshot(`
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: rgb(66, 165, 245);
  color: white;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`);
});
