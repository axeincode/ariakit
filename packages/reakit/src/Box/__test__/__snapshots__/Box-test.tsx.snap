// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled opaque: 
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: rgb(33, 150, 243);;
  color: white;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`;

exports[`styled palette: 
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: rgb(33, 150, 243);
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`;

exports[`styled tone: 
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: rgb(66, 165, 245);
  color: white;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="c0"
/>
`;
